libs/community/langchain_community/graph_vectorstores/extractors/gliner_link_extractor.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Dict, Iterable, List, Optional, Set, Union
+
+from langchain_core.documents import Document
+from langchain_core.graph_vectorstores.links import Link
+
+from langchain_community.graph_vectorstores.extractors.link_extractor import (
+    LinkExtractor,
+)
+
+# TypeAlias is not available in Python 3.9, we can't use that or the newer `type`.
+GLiNERInput = Union[str, Document]
+
+
+class GLiNERLinkExtractor(LinkExtractor[GLiNERInput]):
+    """Link documents with common named entities using GLiNER <https://github.com/urchade/GLiNER>."""
+
+    def __init__(
+        self,
+        labels: List[str],
+        *,
+        kind: str = "entity",
+        model: str = "urchade/gliner_mediumv2.1",
+        extract_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """Extract keywords using GLiNER.
+
+        Example:
+
+            .. code_block: python
+                extractor = GLiNERLinkExtractor(
+                    labels=["Person", "Award", "Date", "Competitions", "Teams"]
+                )
+                results = extractor.extract_one("some long text...")
```

Comment:
The list of example labels is helpful!

To make this even clearer it would be nice to see an example sentence with actual example results. 

The difficult for users is going to be trying to understand what type of transformation this code does (what goes in and what comes out), to figure out when/if they should use it.